src/lib.rs

```diff
@@ -285,30 +316,85 @@ pub fn encode_config_buf<T: ?Sized + AsRef<[u8]>>(input: &T, config: Config, buf
     buf_bytes.resize(orig_buf_len.checked_add(encoded_size)
                          .expect("usize overflow when calculating expanded buffer size"), 0);
 
-    let b64_output = &mut buf_bytes[orig_buf_len..];
+    let mut b64_output = &mut buf_bytes[orig_buf_len..];
 
-    let encoded_bytes = encode_with_padding(input_bytes, b64_output, config.char_set.encode_table(),
-                                            config.pad);
+    encode_with_padding_line_wrap(&input_bytes, &config, encoded_size, &mut b64_output);
+}
 
-    if let LineWrap::Wrap(line_len, line_end) = config.line_wrap {
-        line_wrap(b64_output, encoded_bytes, line_len, line_end);
-    }
+/// Encode arbitrary octets as base64.
+/// Writes into the supplied output buffer.
+///
+/// This is useful if you wish to avoid allocation entirely (e.g. encoding into a stack-resident
+/// or statically-allocated buffer).
+///
+/// # Panics
+///
+/// If `output` is too small to hold the encoded version of `input`, a panic will result.
+///
+/// # Example
+/// 
+/// ```rust
+/// extern crate base64;
+/// 
+/// fn main() {
+///     let s = b"hello internet!";
+///     let mut buf = Vec::new();
+///     // make sure we'll have a slice big enough for base64 + padding
+///     buf.resize(s.len() * 4 / 3 + 4, 0);
+///
+///     let bytes_written = base64::encode_config_slice(s,
+///                             base64::STANDARD, &mut buf);
+///
+///     // shorten our vec down to just what was written
+///     buf.resize(bytes_written, 0);
```

Comment:
Is this necessary? I'm not actually sure what it accomplishes.

Reply:
We made the `Vec` conservatively sized originally, so if we didn't truncate it back down, we'd have some stray 0 bytes that would prevent the subsequent `decode` from working.

Reply:
Oh right I always seem to forget (when working with b64 in particular, for some reason) that you can't have uninitialized data in a slice.

Reply:
Yeah, it's a bit awkward -- I wish there was a type-safe way to say "give me some data that I'm about to overwrite". Fortunately modern CPUs do a great job of splatting 0 into memory.